hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -80,11 +206,69 @@ sealed abstract class Block extends Product with AutoLocated:
     case Assign(lhs, rhs, rest) => Set(lhs) ++ rhs.freeVars ++ rest.freeVars
     case AssignField(lhs, nme, rhs, rest) => lhs.freeVars ++ rhs.freeVars ++ rest.freeVars
     case Define(defn, rest) => defn.freeVars ++ rest.freeVars
-    case HandleBlock(lhs, res, cls, handlers, body, rest) =>
-      (body.freeVars - lhs) ++ rest.freeVars ++ handlers.flatMap(_.freeVars)
+    case HandleBlock(lhs, res, par, cls, hdr, bod, rst) =>
+      (bod.freeVars - lhs) ++ rst.freeVars ++ hdr.flatMap(_.freeVars)
     case HandleBlockReturn(res) => res.freeVars
     case End(msg) => Set.empty
 
+  def floatOutDefns =
+    def rec(b: Block, acc: List[Defn]): (Block, List[Defn]) =
+      b match
```

Comment:
Pro-tip: you don't need to indent after `match`. For nested matches (like here), this can save a lot on of needless indentation.